v-next/hardhat/src/types/network.ts

```diff
@@ -65,3 +71,9 @@ export interface NetworkConnection<
 
   close(): Promise<void>;
 }
+
+export interface JsonRpcServer {
+  listen(): Promise<{ address: string; port: number }>;
+  waitUntilClosed(): Promise<void>;
+  close(): Promise<void>;
+}
```

Comment:
Can I check that it is our intention to expose `waitUntilClosed`? We use it in one place the `npx hardhat node` task to ensure everything is fully closed before finishing the task.

Reply:
Good point, I'll remove it from the interface, I don't think it's necessary

Reply:
> I think that is too alike (I almost said close)

Agreed.

TBH, I have no strong preference on the actual name. `afterClosed()` works for me too. `waitUntilClosed()` with good jsdocs too.